### PR TITLE
felix/bpf: fix for WG no longer uses failsafes

### PR DESCRIPTION
**Pull request by tomastigera**
Previously, WG traffic was allowed by patching failsafe ports. It was broken by https://github.com/projectcalico/calico/pull/6250

We allow wireguard traffic between our nodes even if a user accidentally
block the traffic by a HEP policy.

We do that only if there was an actuall DENY by a bad policy to save
ourselves constantly testing whether to let it pass or not at the fast
path. The test requires 2 RT lookups.

When WG traffic is blocked by a bad policy, we allow it and only then
create conntrack entry to allow subsequent packets. Those entries are
kept alive by constant traffic (most of the time) or recreated after
they expire after policy is applied again.

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
ebpf: WG traffic is allowed by the HEP programs in case of a conflicting policy.
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

**Review by mazdakn — APPROVED** [review]
LGTM

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Pull request merged**
